### PR TITLE
Patch for JDK-9

### PR DESCRIPTION
**Pull request by MartyIX**
Hi @epickrram, 

This is my patch applied to the current master of -Core (https://github.com/OpenHFT/Chronicle-Core/issues/15#issuecomment-326003962).

Regards,
Martin

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/core/Maths.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 package net.openhft.chronicle.core;
 
+import net.openhft.chronicle.core.annotation.Java9;
```
Comment (epickrram) [inline src/main/java/net/openhft/chronicle/core/Maths.java]: This import doesn't seem to be used anywhere - can it be removed?

File: src/main/java/net/openhft/chronicle/core/Maths.java [hunk 2]
```diff
@@ -285,17 +286,31 @@ public static long hash64(@NotNull CharSequence cs) {
 
     public static long hash64(@NotNull String s) {
```
Comment (epickrram) [inline src/main/java/net/openhft/chronicle/core/Maths.java]: Please add a unit test to MathsTest that exercises these code paths.

File: src/main/java/net/openhft/chronicle/core/util/StringUtils.java [hunk 3]
```diff
@@ -99,11 +99,20 @@ public static boolean isEqual(@Nullable StringBuilder s, @Nullable CharSequence
         if (cs == null) return false;
         int length = cs.length();
         if (s.length() != length) return false;
-        char[] chars = StringUtils.extractChars(s);
-        for (int i = 0; i < length; i++)
-            if (chars[i] != cs.charAt(i))
-                return false;
-        return true;
+
```
Comment (epickrram) [inline src/main/java/net/openhft/chronicle/core/util/StringUtils.java]: Please add a unit test to StringUtilsTest to exercise these code paths.

File: src/main/java/net/openhft/chronicle/core/Maths.java (hunk 1 above)
Comment (MartyIX) [reply src/main/java/net/openhft/chronicle/core/Maths.java]: You are right. Fixed.

File: src/main/java/net/openhft/chronicle/core/Maths.java (hunk 2 above)
Comment (MartyIX) [reply src/main/java/net/openhft/chronicle/core/Maths.java]: @epickrram https://github.com/OpenHFT/Chronicle-Core/pull/37/files#diff-b2162a42c901ba85c335d409e8a97350R323 - Does the test look alright to you?

Comment (epickrram) [reply src/main/java/net/openhft/chronicle/core/Maths.java]: Looks good.

File: src/main/java/net/openhft/chronicle/core/util/StringUtils.java (hunk 3 above)
Comment (MartyIX) [reply src/main/java/net/openhft/chronicle/core/util/StringUtils.java]: @epickrram I have added the test.

File: src/test/java/net/openhft/chronicle/core/MathsTest.java [hunk 4]
```diff
@@ -317,4 +318,42 @@ public void testHashStringBuilderFromInterner() throws Exception {
...
+
+        String a2 = new StringBuilder().append("T").append("e").toString() + "st";
+        long ah2 = Maths.hash64(a2);
+
+        assertEquals(ah1, ah2);
+
+        // UTF8 & Equality test
+        String u1 = "€";
+        long uh1 = Maths.hash64(u1);
+        assertEquals(-11958288497246124L, uh1);
```
Comment (epickrram) [inline src/test/java/net/openhft/chronicle/core/MathsTest.java]: This fails when running under java 8. Can you take a look at fixing pls?

**Comment by MartyIX** [comment]
Fixed. Sorry for that.

**Pull request merged**